api/v1beta2/types.go

```diff
@@ -142,6 +142,21 @@ var (
 	ResourceTypeResourceGroup = ResourceType("resourceGroup")
 )
 
+// COSInstanceReference represents an IBM Cloud COS instance.
+type COSInstanceReference struct {
+	// id is the ID of the COS instance.
+	// +required
+	ID *string `json:"id"`
+
+	// bucketID is the IBM Cloud COS bucket ID.
+	// +optional
+	BucketID *string `json:"bucketID,omitempty"`
+
```

Comment:
So if we want to reuse the same struct for PowerVS as well then lets add BucketName and Name here. 
So this allows us to create and use the COSinstance and Bucket with user provided name.

Just for clarification, In PowerVS we create these resources to upload the ignition data.

Reply:
I plan to handle the process for VSI Custom Image using a VPC unique type instead, and since Ignition data management is expected to occur elsewhere, I don't think COS details like this are necessary. I will divert from the PowerVS path in respect to COS.

I will be removing this `COSInstanceReference` type as I don't expect to need it for VPC based on my findings.